src/App.tsx

```diff
@@ -1,173 +1,132 @@
-import { Loader } from './components/Loader';
-
+import {
+  Routes,
+  Route,
+  Navigate,
+  NavLink,
+} from 'react-router-dom';
 import './App.scss';
-
-export const App = () => (
-  <div data-cy="app">
-    <nav
-      data-cy="nav"
-      className="navbar is-fixed-top has-shadow"
-      role="navigation"
-      aria-label="main navigation"
-    >
-      <div className="container">
-        <div className="navbar-brand">
-          <a className="navbar-item" href="#/">
-            Home
-          </a>
-
-          <a
-            className="navbar-item has-background-grey-lighter"
-            href="#/people"
-          >
-            People
-          </a>
-        </div>
-      </div>
-    </nav>
-
-    <main className="section">
-      <div className="container">
-        <h1 className="title">Home Page</h1>
-        <h1 className="title">People Page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="block">
-          <div className="box table-container">
-            <Loader />
-
-            <p data-cy="peopleLoadingError" className="has-text-danger">
-              Something went wrong
-            </p>
-
-            <p data-cy="noPeopleMessage">
-              There are no people on the server
-            </p>
-
-            <table
-              data-cy="peopleTable"
-              className="table is-striped is-hoverable is-narrow is-fullwidth"
+import { useCallback, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { HomePage } from './components/HomePage/HomePage';
+import { PeoplePage } from './components/PeoplePage/PeoplePage';
+import { getPeople } from './api';
+import { Person } from './types';
+
+export const App = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isDataUploaded, setIsDataUploaded] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then(uploadedPeople => {
+        setPeople(uploadedPeople);
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsDataUploaded(true);
+      });
+  }, []);
+
+  const findMotherSlug = useCallback((child: Person): string | null => {
```

Comment:
`useCallback` and `useMemo` are also functions with some logic, so we need to think twice before we start using them. Try to refactor your code and leave only important `useCallback`




`useCallback` we use to prevent rendering for memoized component because we don't create a new link on this function

 `useMemo` we use to prevent rendering for memoized component because we don't create a new link on array or object, and the second reason is to prevent some hard operation of array or object


memoized component - a component wrapped with `React.memo`